### PR TITLE
Add `eksctl utils update-cluster-vpc-config` to update the cluster VPC config

### PR DESCRIPTION
**Pull request by cPu1**
### Description

There are two utils commands `update-cluster-endpoints` and `set-public-access-cidrs` that allow updating the VPC config. Both commands use the same underlying API call `UpdateClusterConfig`, passing in the changed fields in `ResourcesVpcConfig`. This changelist adds a new `update-cluster-vpc-config` command mirroring the `ResourcesVpcConfig` field in the `UpdateClusterConfig` API call. It provides the functionality of `update-cluster-endpoints` and `set-public-access-cidrs` in a new command `eksctl utils update-cluster-vpc-config`. In a future PR, the existing commands will be deprecated. 

To preserve the behaviour of the existing commands whose functionality is being added to `update-cluster-vpc-config`, it  also supports the CLI options from those commands. A command that passes all supported CLI options looks like:

```shell
$ eksctl utils update-cluster-vpc-config —public-access-cidrs=<> —private-access —public-access
```
Please note that `set-public-access-cidrs` does not accept any CLI option, but instead treats its first argument as a set of public access CIDRs, e.g., `eksctl utils set-public-access-cidrs 1.1.1.1/32,2.2.2.2/32`, so a new CLI option `set-public-access-cidrs` is being introduced for `update-cluster-vpc-config`.

To avoid any confusion as to which fields under `vpc` can be updated when using the config file variant, `update-cluster-vpc-config` prints a warning listing the fields that can be updated.

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 1]
```diff
@@ -954,6 +954,9 @@ type OutpostInfo interface {
 	GetOutpost() *Outpost
 }
 
+// ErrUnsupportedLocalCluster is an error for when an unsupported operation is attempted on a local cluster.
+var ErrUnsupportedLocalCluster = errors.New("this operation is not supported on local clusters")
```
Comment (a-hilaly) [inline pkg/apis/eksctl.io/v1alpha5/types.go:958]: Can we mention in the error that these are private clusters/outpost?

**Review by a-hilaly — APPROVED** [review]
Niice, I learned a few things about eksctl internal by reading this PR 👍

File: pkg/apis/eksctl.io/v1alpha5/types.go (hunk 1 above)
Comment (cPu1) [reply pkg/apis/eksctl.io/v1alpha5/types.go:961]: I haven't changed the wording of the error message in this PR, but it makes sense to include Outposts in the error message.

**Review by TiberiuGC — APPROVED** [review]

**Comment by rfarro82** [comment]
I ran into an issue when using this version and the new command. Ple
```

**Comment by rfarro82** [comment]
I ran into an issue when using 0.164.0 version with the new command. No changes are being made to the cidrs or endpoints. Reverted back to 0.163.0 and old commands and working again. 
```
+ eksctl upgrade cluster -f clusters/cluster.yaml --approve
2023-11-03 16:04:42 [!]  NOTE: cluster VPC (subnets, routing & NAT Gateway) configuration changes are not yet implemented
2023-11-03 16:04:42 [ℹ]  no cluster version update required
2023-11-03 16:04:43 [ℹ]  re-building cluster stack "eksctl-cluster"
2023-11-03 16:04:43 [✔]  all resources in cluster stack "eksctl-cluster" are up-to-date
2023-11-03 16:04:43 [ℹ]  checking security group configuration for all nodegroups
2023-11-03 16:04:43 [ℹ]  all nodegroups have up-to-date cloudformation templates
+ eksctl utils update-cluster-vpc-config -f clusters/cluster.yaml --approve
2023-11-03 16:04:43 [ℹ]  only changes to vpc.clusterEndpoints, vpc.publicAccessCIDRs, vpc.controlPlaneSubnetIDs and vpc.controlPlaneSecurityGroupIDs are updated in the EKS API, changes to any other fields will be ignored
2023-11-03 16:04:44 [ℹ]  using region us-east-1
2023-11-03 16:04:44 [✔]  Kubernetes API endpoint access for cluster "cluster" in "us-east-1" is already up-to-date
2023-11-03 16:04:44 [ℹ]  current public access CIDRs: [$CIDRS]
2023-11-03 16:04:44 [ℹ]  will update public access CIDRs for cluster "dev" in "us-east-1" to: [$CIDRS]
Error: error updating CIDRs for public access: operation error EKS: UpdateClusterConfig, https response error StatusCode: 400, RequestID: ##########, InvalidParameterException: Cluster is already at the desired configuration with endpointPrivateAccess: true , endpointPublicAccess: true, and Public Endpoint Restrictions: [$CIDRS]
```

**Comment by cPu1** [comment]
> I ran into an issue when using 0.164.0 version with the new command. No changes are being made to the cidrs or endpoints. Reverted back to 0.163.0 with old commands and working again.
> 
> ```
> + eksctl upgrade cluster -f clusters/cluster.yaml --approve
> 2023-11-03 16:04:42 [!]  NOTE: cluster VPC (subnets, routing & NAT Gateway) configuration changes are not yet implemented
> 2023-11-03 16:04:42 [ℹ]  no cluster version update required
> 2023-11-03 16:04:43 [ℹ]  re-building cluster stack "eksctl-cluster"
> 2023-11-03 16:04:43 [✔]  all resources in cluster stack "eksctl-cluster" are up-to-date
> 2023-11-03 16:04:43 [ℹ]  checking security group configuration for all nodegroups
> 2023-11-03 16:04:43 [ℹ]  all nodegroups have up-to-date cloudformation templates
> + eksctl utils update-cluster-vpc-config -f clusters/cluster.yaml --approve
> 2023-11-03 16:04:43 [ℹ]  only changes to vpc.clusterEndpoints, vpc.publicAccessCIDRs, vpc.controlPlaneSubnetIDs and vpc.controlPlaneSecurityGroupIDs are updated in the EKS API, changes to any other fields will be ignored
> 2023-11-03 16:04:44 [ℹ]  using region us-east-1
> 2023-11-03 16:04:44 [✔]  Kubernetes API endpoint access for cluster "cluster" in "us-east-1" is already up-to-date
> 2023-11-03 16:04:44 [ℹ]  current public access CIDRs: [$CIDRS]
> 2023-11-03 16:04:44 [ℹ]  will update public access CIDRs for cluster "dev" in "us-east-1" to: [$CIDRS]
> Error: error updating CIDRs for public access: operation error EKS: UpdateClusterConfig, https response error StatusCode: 400, RequestID: ##########, InvalidParameterException: Cluster is already at the desired configuration with endpointPrivateAccess: true , endpointPublicAccess: true, and Public Endpoint Restrictions: [$CIDRS]
> ```

@rfarro82 could you share the CIDR elements from both your config file and by making a call to `aws eks describe-cluster` after replacing them with non-sensitive values (e.g., x, y, z)?

**Comment by rfarro82** [comment]
The aws eks describe-cluster command and replacing the sensitive values with fake values. 
```
            "endpointPublicAccess": true,
            "endpointPrivateAccess": true,
            "publicAccessCidrs": [
                "38.0.1.2/32",
                "34.0.1.2/24",
                "199.0.1.2/32",
                "34.1.2.3/28",
                "52.0.1.2/32"
            ]
        },
```
Cluster Config file
```
clusterEndpoints:
    publicAccess:  true
    privateAccess: true
  publicAccessCIDRs: [
    "38.0.1.2/32", 
    "199.0.1.2/32", 
    "52.0.1.2/32", 
    "34.1.2.3/28",  
    "34.0.1.2/24", 
    ]
```
Observing order in list does not match up.

**Comment by nipr-jdoenges** [comment]
> I ran into an issue when using 0.164.0 version with the new command. No changes are being made to the cidrs or endpoints. Reverted back to 0.163.0 with old commands and working again.
> 
> ```
> + eksctl upgrade cluster -f clusters/cluster.yaml --approve
> 2023-11-03 16:04:42 [!]  NOTE: cluster VPC (subnets, routing & NAT Gateway) configuration changes are not yet implemented
> 2023-11-03 16:04:42 [ℹ]  no cluster version update required
> 2023-11-03 16:04:43 [ℹ]  re-building cluster stack "eksctl-cluster"
> 2023-11-03 16:04:43 [✔]  all resources in cluster stack "eksctl-cluster" are up-to-date
> 2023-11-03 16:04:43 [ℹ]  checking security group configuration for all nodegroups
> 2023-11-03 16:04:43 [ℹ]  all nodegroups have up-to-date cloudformation templates
> + eksctl utils update-cluster-vpc-config -f clusters/cluster.yaml --approve
> 2023-11-03 16:04:43 [ℹ]  only changes to vpc.clusterEndpoints, vpc.publicAccessCIDRs, vpc.controlPlaneSubnetIDs and vpc.controlPlaneSecurityGroupIDs are updated in the EKS API, changes to any other fields will be ignored
> 2023-11-03 16:04:44 [ℹ]  using region us-east-1
> 2023-11-03 16:04:44 [✔]  Kubernetes API endpoint access for cluster "cluster" in "us-east-1" is already up-to-date
> 2023-11-03 16:04:44 [ℹ]  current public access CIDRs: [$CIDRS]
> 2023-11-03 16:04:44 [ℹ]  will update public access CIDRs for cluster "dev" in "us-east-1" to: [$CIDRS]
> Error: error updating CIDRs for public access: operation error EKS: UpdateClusterConfig, https response error StatusCode: 400, RequestID: ##########, InvalidParameterException: Cluster is already at the desired configuration with endpointPrivateAccess: true , endpointPublicAccess: true, and Public Endpoint Restrictions: [$CIDRS]
> ```

Anyone coming across this, it appears https://github.com/eksctl-io/eksctl/pull/7483 may have addressed this issue.

**Pull request merged**